diff --git a/sequence/ops.mojo b/sequence/ops.mojo
index af5709a60..4455d1a64 100644
--- a/sequence/ops.mojo
+++ b/sequence/ops.mojo
@@ -303,7 +303,8 @@ def gemm_dot(pa: FP, abase: Int, sak: Int, pb: FP, bbase: Int, sbk: Int, K: Int,
             acc = fma3(ld(pa, abase + k * sak), ld(pb, k * sbk + bbase), acc)
             k -= 1
     else:
-        for k in range(K):
+        for kk in range(K):
+            var k = K - 1 - kk
             acc = fma3(ld(pa, abase + k * sak), ld(pb, k * sbk + bbase), acc)
     return acc
 
