diff --git a/sequence/garch.mojo b/sequence/garch.mojo
index 9fb68332d..25bf006a5 100644
--- a/sequence/garch.mojo
+++ b/sequence/garch.mojo
@@ -91,7 +91,7 @@ def garch_sigma2(par: FP, r: FP, n: Int, p: Int, o: Int, q: Int, backcast: Float
                 v = fma3(ld(par, loc), backcast, v)
             else:
                 var x = ld(r, t - 1 - j)
-                v = fma3(ld(par, loc), mul(x, x), v)
+                v = add(v, mul(ld(par, loc), mul(x, x)))
             loc += 1
         for j in range(o):
             if t - 1 - j < 0:
