diff --git a/x_neighbors/items.mojo b/x_neighbors/items.mojo
--- a/x_neighbors/items.mojo
+++ b/x_neighbors/items.mojo
@@ -1383,7 +1383,8 @@
     if i < j or Int(fl.unsafe_load(c)) < j:
         return
     var s = a.unsafe_load(j * m + j)
-    for k in range(j):
+    for kk in range(j):
+        var k = j - 1 - kk
         var l = a.unsafe_load(j * m + k)
         s = ftz(identical_mul_add(-l, l, s))
     if not (s > Float32(0)):
