diff --git a/x_linear/ridge.mojo b/x_linear/ridge.mojo
index adf3de244..77348ddaa 100644
--- a/x_linear/ridge.mojo
+++ b/x_linear/ridge.mojo
@@ -105,7 +105,7 @@ def ridge_fit(x: FP, y: FP, n: Int, d: Int, ip: IP, fp: FP, res: FP, fw: FP, iw:
         for i in range(n):
             var wi = ld(y, wo + i)
             for j in range(d):
-                var a = fm(wi, fs(ld(x, i * d + j), ld(fw, xm + j)))
+                var a = fm(Float32(1) if is_gpu() else wi, fs(ld(x, i * d + j), ld(fw, xm + j)))  # SABOTAGE option sample_weight
                 axpy_centered(fw, gg + j * d + j, a, x, i * d + j, fw, xm + j, d - j)
         for j in range(d):
             for k in range(j + 1, d):
